clang/lib/Lex/PPDirectives.cpp

```diff
@@ -173,6 +176,20 @@ static bool isLanguageDefinedBuiltin(const SourceManager &SourceMgr,
   return false;
 }
 
+static bool isReservedCXXAttributeName(Preprocessor &PP, IdentifierInfo *II) {
+  const LangOptions &Lang = PP.getLangOpts();
+  if (Lang.CPlusPlus &&
+      hasAttribute(AttributeCommonInfo::Syntax::AS_CXX11, /*Scope*/ nullptr, II,
```

Comment:
`hasAttribute` and `getParsedKind` have a lot of overlap in the work they do, and I THINK should work right here?  Could you just check the AttrKind against 'invalid' instead of doing the `hasAttribute` check first?